server/controllers/dashboard/stats.js

```diff
@@ -58,28 +58,27 @@ function invoiceStat(req, res, next) {
 
   // query invoices
   const sqlBalance =
-    `SELECT 
-    (debit - credit) as balance, project_id, cost
+    `SELECT (debit - credit) as balance, project_id, cost
      FROM (
-      SELECT 
-        SUM(debit_equiv) as debit, SUM(credit_equiv) as credit, invoice.project_id, invoice.cost
-      FROM 
-        combined_ledger
-      JOIN 
-        invoice ON combined_ledger.record_uuid = invoice.uuid OR 
-        combined_ledger.reference_uuid = invoice.uuid
-      WHERE 
-        invoice.uuid NOT IN (
-          SELECT 
-            voucher.reference_uuid 
-          FROM voucher 
-          WHERE voucher.type_id = ${CANCELED_TRANSACTION_TYPE})
+      (
+        SELECT SUM(debit_equiv) as debit, SUM(credit_equiv) as credit, invoice.project_id, invoice.cost
+        FROM posting_journal
+        JOIN invoice ON posting_journal.record_uuid = invoice.uuid OR posting_journal.reference_uuid = invoice.uuid
+        WHERE invoice.uuid NOT IN (SELECT voucher.reference_uuid FROM voucher WHERE voucher.type_id = ${CANCELED_TRANSACTION_TYPE})
```

Comment:
You should replace this `WHERE invoice.uuid NOT IN()` with `WHERE invoice.reversed = 0`.  It will be much faster :+1: